libs/wingsdk/src/target-sim/secret.inflight.ts

```diff
@@ -35,10 +45,17 @@ export class Secret implements ISecretClient, ISimulatorResourceInstance {
       type: TraceType.RESOURCE,
       timestamp: new Date().toISOString(),
     });
-    return this.secretValue;
+
+    const secrets = JSON.parse(fs.readFileSync(this.secretsFile, "utf-8"));
+
+    if (!secrets[this.name]) {
+      throw new Error(`No secret value for secret ${this.name}`);
+    }
+
+    return secrets[this.name];
   }
 
-  async valueJson(): Promise<Json> {
+  public async valueJson(): Promise<Json> {
```

Comment:
Question: will be it possible at some point to do this here?

```ts
public async valueJson<T>(): Promise<T>
```

Reply:
I think we're planning to eventually add support for generics so you can represent types like `cloud.Queue<UserStruct>`, `cloud.Function<Input, Output>`, or `cloud.Secret<T>`, and perform operations safely based those parameterized types -- but we don't have it scheduled on our short term roadmap (next few weeks). I think it would be really powerful to have though. Here's our tracking issue: https://github.com/winglang/wing/issues/104